### PR TITLE
DP-15448 add slackalerts module

### PR DESCRIPTION
**Pull request by a-username**
Add the slackalerts module and switches to using a topic_map variable.

### CODE REVIEW

File: slackalerts/main.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      "Name" = var.name
+    },
+  )
+}
+
+resource "aws_sns_topic_subscription" "default" {
+  count     = var.sns_topic_count
+  endpoint  = aws_lambda_function.sns_to_slack.arn
+  protocol  = "lambda"
+  topic_arn = element(var.sns_topics, count.index)
```
Comment (rbayliss) [inline slackalerts/main.tf:30]: I think you did the right thing here in keeping this change as minimal as possible.  Eventually though, I'd love to get rid of the `sns_topics` and `sns_topic_count`, and `default_channel` variables entirely, and just rely on `topic_map`.  Eg:
```hcl
locals {
  sns_topic_arns = [for topic in var.topic_map: topic.arn]
}

resource "aws_sns_topic_subscription" "default" {
  count     = length(local.sns_listen_topics)
  endpoint  = aws_lambda_function.sns_to_slack.arn
  protocol  = "lambda"
  topic_arn = element(local.sns_topic_arns, count.index)
}
```
That way, you'd use that `topic_map` variable to control both the SNS topics we're listening to, and the Slack settings for the topic.  Simple, explicit, and clean. But again - not a change we need to make right now.

File: slackalerts/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+variable "topic_map" {
+  description = "SNS topics mapped to Slack channels."
+  type = list(object({
+    topic_arn = string
+    channel = string
+    username = string
+    icon_emoji = string
+  }))
```
Comment (rbayliss) [inline slackalerts/variables.tf:39]: Can we provide a default value here of `[]`?

**Review by rbayliss — CHANGES_REQUESTED** [review]
Looking good - one small change requested, then I think we can merge and tag on this side.

File: slackalerts/main.tf (hunk 1 above)
Comment (a-username) [reply slackalerts/main.tf:30]: Ah! I wasn't sure what to do with these, but I'll leave a comment in the ticket and we can make a follow-up if that makes sense?

File: slackalerts/variables.tf (hunk 2 above)
Comment (a-username) [reply slackalerts/variables.tf:39]: all set!

**Comment by rbayliss** [comment]
Looks great, thank you!

**Pull request merged**